src/draw.js

```diff
@@ -64,24 +67,26 @@ function not(f) {
 }
 
 function addCircle(drawable, key) {
+  var drawColor = drawable.current.drawColor;
   var sameCircle = function(s) {
-    return s.length === 1 && s[0] === key;
+    return s.drawColor === drawColor && s.orig === key && !s.dest;
```

Comment:
Supports drawing all 4 different circle colors on the same square inside each other, in contrast to arrows that only allow 1 color.